src/Plugin.php

```diff
@@ -441,9 +441,9 @@ private function getMaxDepth()
             $maxDepth = $extra[self::KEY_MAX_DEPTH];
             $minDepth = $this->getMinDepth();
 
-            if (is_int($maxDepth) === false     /* Must be an integer */
-                || $maxDepth <= $minDepth       /* Larger than the minimum */
-                || is_float($maxDepth) === true /* Within the boundaries of integer */
+            if ((string) (int) $maxDepth !== (string) $maxDepth /* Must be an integer or cleanly castable to one */
```

Comment:
I've never seen this before... Is there any benefit of this over using [`is_numeric`](https://www.php.net/manual/en/function.is-numeric.php)?

Reply:
Yes, @Potherca. `is_numeric` would match floating point numbers, whereas we only want to match integers and string representations of them. There's no builtin function for that, as you know, but typecasting a string to an integer and back and then seeing whether it matches the original value is a common idiom for it. It's probably it's fairly intuitive once you've been introduced to it. :)